app/store/FavouriteStore.js

```diff
@@ -209,12 +215,20 @@ export default class FavouriteStore extends Store {
    * and on fail callback function under onFail key
    */
   saveFavourite(actionData) {
-    const { onFail, ...data } = actionData;
+    let { ...data } = actionData;
+    const { onFail } = actionData;
     if (typeof data !== 'object') {
       onFail();
       throw new Error(`New favourite is not a object:${JSON.stringify(data)}`);
     }
     this.fetchingOrUpdating();
+    if (data.type === 'bikeStation') {
+      data = {
+        ...data,
+        stationId: data.stationId.split(':')[1],
+        networks: [data.networks],
```

Comment:
Where we call this function the data field should be called network, not networks and then here we store it in networks field as array.